crosscat/utils/inference_utils.py

```diff
@@ -124,30 +118,45 @@ def calculate_MI_bounded_discrete(X, Y, M_c, X_L, X_D):
         component_models_X[i] = cluster_models[X]
         component_models_Y[i] = cluster_models[Y]
 
-    MI = 0.0
+    def marginal_predictive_by_cluster(value, component_models):
+        return numpy.array([
+            component_models[j].calc_element_predictive_logp(value)
```

Comment:
I feel like either we should drop the _logp and _logps suffixes everywhere, or else use them everywhere. The mix is a tad confusing.